lib/Generate.js

```diff
@@ -414,4 +339,110 @@ function generateMainModule(
   return mainFile;
 }
 
+function makeTestFileBody(
+  testModules /*: Array<{
+    moduleName: string,
+    possiblyTests: Array<string>,
+  }> */,
+  optsCode /*: string */
+) /*: string */ {
+  const imports = testModules.map((mod) => `import ${mod.moduleName}`);
+
+  const possiblyTestsList = makeList(testModules.map(makeModuleTuple));
+
+  return `
+${imports.join('\n')}
+
+import Test.Reporter.Reporter exposing (Report(..))
+import Console.Text exposing (UseColor(..))
+import Test.Runner.Node
+import Test
+
+main : Test.Runner.Node.TestProgram
+main =
+    Test.Runner.Node.run
+        ${indentAllButFirstLine('        ', optsCode)}
+        ${indentAllButFirstLine('        ', possiblyTestsList)}
+  `.trim();
+}
+
+function makeModuleTuple(mod /*: {
+  moduleName: string,
+  possiblyTests: Array<string>,
+} */) /*: string */ {
+  const list = mod.possiblyTests.map(
+    (test) => `Test.Runner.Node.check ${mod.moduleName}.${test}`
+  );
+
+  return `
+( "${mod.moduleName}"
+, ${indentAllButFirstLine('  ', makeList(list))}
+)
+  `.trim();
+}
+
+function makeList(parts /*: Array<string> */) /*: string */ {
+  const list = parts.map(
+    (part, index) =>
+      `${index === 0 ? '' : ', '}${indentAllButFirstLine('  ', part)}`
+  );
+
+  return `
+[ ${list.join('\n')}
+]
+  `.trim();
+}
+
+function indentAllButFirstLine(indent, string) {
+  return string
+    .split('\n')
+    .map((line, index) => (index === 0 ? line : indent + line))
+    .join('\n');
+}
+
+function makeOptsCode(
+  fuzz /*: number */,
+  seed /*: number */,
+  report /*: string */,
+  testFileGlobs /*: Array<string> */,
+  testFilePaths /*: Array<string> */,
+  processes /*: number */
+) /*: string */ {
+  // TODO: CLI args should be parsed, validated and defaulted properly in elm-test.js.
+  const finalSeed = isNaN(seed)
+    ? Math.floor(Math.random() * 407199254740991) + 1000
+    : seed;
```

Comment:
It’s super weird to create the default seed deep into the code generation routine. But this is out of scope for this PR.

Reply:
I have edited the PR description with a follow up task to make an issue to move it :+1:

Reply:
Possibly related: https://github.com/rtfeldman/node-test-runner/issues/248

Reply:
Also related: https://github.com/rtfeldman/node-test-runner/issues/211